src/ngraph/pass/constant_folding.cpp

```diff
@@ -763,3 +766,176 @@ void pass::ConstantFolding::construct_constant_quantize()
+    case element::Type_t::u8:
+        return fold_constant_convert_helper1<TI, uint8_t>(constant, output_element_type);
+    case element::Type_t::u16:
+        return fold_constant_convert_helper1<TI, uint16_t>(constant, output_element_type);
+    case element::Type_t::u32:
+        return fold_constant_convert_helper1<TI, uint32_t>(constant, output_element_type);
+    case element::Type_t::u64:
+        return fold_constant_convert_helper1<TI, uint64_t>(constant, output_element_type);
+    }
+#pragma GCC diagnostic pop
+}
+
+shared_ptr<op::Constant> fold_constant_convert(shared_ptr<op::Constant> constant,
+                                               const element::Type& output_element_type)
+{
+    auto& input_element_type = constant->get_output_element_type(0);
+
+    if (input_element_type == output_element_type)
+    {
+        return constant;
+    }
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic error "-Wswitch"
+#pragma GCC diagnostic error "-Wswitch-enum"
+    switch (input_element_type.get_type_enum())
+    {
+    case element::Type_t::undefined:
+        NGRAPH_CHECK(false, "Encountered 'undefined' element type in fold_constant_convert");
+        break;
+    case element::Type_t::dynamic:
+        NGRAPH_CHECK(false, "Encountered 'dynamic' element type in fold_constant_convert");
+        break;
+    case element::Type_t::boolean:
+        return fold_constant_convert_helper0<char>(constant, output_element_type);
+    case element::Type_t::bf16:
+        return fold_constant_convert_helper0<bfloat16>(constant, output_element_type);
+    case element::Type_t::f16:
+        return fold_constant_convert_helper0<float16>(constant, output_element_type);
+    case element::Type_t::f32:
+        return fold_constant_convert_helper0<float>(constant, output_element_type);
+    case element::Type_t::f64:
+        return fold_constant_convert_helper0<double>(constant, output_element_type);
+    case element::Type_t::i8:
+        return fold_constant_convert_helper0<int8_t>(constant, output_element_type);
+    case element::Type_t::i16:
+        return fold_constant_convert_helper0<int16_t>(constant, output_element_type);
+    case element::Type_t::i32:
+        return fold_constant_convert_helper0<int32_t>(constant, output_element_type);
+    case element::Type_t::i64:
+        return fold_constant_convert_helper0<int64_t>(constant, output_element_type);
+    case element::Type_t::u8:
+        return fold_constant_convert_helper0<uint8_t>(constant, output_element_type);
+    case element::Type_t::u16:
+        return fold_constant_convert_helper0<uint16_t>(constant, output_element_type);
+    case element::Type_t::u32:
+        return fold_constant_convert_helper0<uint32_t>(constant, output_element_type);
+    case element::Type_t::u64:
+        return fold_constant_convert_helper0<uint64_t>(constant, output_element_type);
+    }
+#pragma GCC diagnostic pop
```

Comment:
GCC 4.8.X has some missed with handling some pragmas. Workaround shown in https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/op/constant.hpp#L311-L313 is needed.